budgetApp1/budgetApp1/BudgetViewController.swift

```diff
@@ -9,49 +9,73 @@ import UIKit
+    private let uiSpecs = UISpecs()
+    
+    // MARK: - IBOutlets
+    @IBOutlet private weak var budgetNameLabel: UILabel!
+    @IBOutlet private weak var endDateLabel: UILabel!
+    @IBOutlet private weak var startDateLabel: UILabel!
+    @IBOutlet weak var toDateLabel: UILabel!
+    @IBOutlet weak var recentTransactionsLabel: UILabel!
+    @IBOutlet weak var tableView: UITableView!
     
-    @IBOutlet weak var budgetNameLabel: UILabel!
-    @IBOutlet weak var endDateLabel: UILabel!
-    @IBOutlet weak var startDateLabel: UILabel!
+    private lazy var budgetViewModel = BudgetViewModel(repository: BudgetRepository(), delegate: self)
+    private lazy var transactionViewModel = AccountTransactionsViewModel(repository: TransactionRepository(), delegate: self)
     
+    // MARK: - Functions
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        viewModel = BudgetViewModel(repository: BudgetRepository(), delegate: self)
-        viewModel?.getBudgets()
-        // Do any additional setup after loading the view.
-         
+        budgetViewModel.fetchBudgets()
+        transactionViewModel.fetchTransactions()
+        setupTableView()
     }
 
-    private func fillUI() {
-        if !isViewLoaded {
-            return
-        }
-        
-        guard let viewModel = viewModel, let budgetList = viewModel.budgetList, !budgetList.isEmpty else {
-            return
-        }
-        budgetNameLabel.text = budgetList.first?.name
-        startDateLabel.text = budgetList.first?.firstMonth
-        endDateLabel.text = budgetList.first?.lastMonth
+    private func updateView(withBudget budget: Budget) {
+        budgetNameLabel.text = budget.name
+        startDateLabel.text = budget.firstMonth
+        endDateLabel.text = budget.lastMonth
+        toDateLabel.text = "–"
+        recentTransactionsLabel.text = "Most recent transactions"
     }
+    
+    private func setupTableView() {
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.register(TransactionTableViewCell.nib(), forCellReuseIdentifier: NibIdentifiers.TransactionViewCellIdentifier)
+        tableView.separatorStyle = .singleLine
+        tableView.separatorColor = uiSpecs.primaryColourOne
+        tableView.separatorInset = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 16)
+    }
+}
 
+extension BudgetViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        60.0
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        transactionViewModel.transactionListCount
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: NibIdentifiers.TransactionViewCellIdentifier) as? TransactionTableViewCell else { return UITableViewCell() }
```

Comment:
indenting off?

Reply:
not entirely, but I indented again to make sure